span.go

```diff
@@ -469,6 +469,8 @@ func setSamplingPriority(s *Span, value interface{}) bool {
 	if !ok {
 		return false
 	}
+	s.RLock()
+	defer s.RUnlock()
```

Comment:
Could we just retrieve the context instead? The samplingState is shared even if the context is overwritten on the span.

Reply:
But getting RLock is ok too.